redhat-developer/codeready-workspaces
# Provide Devfile v2 for che-idea editor

### What does this PR do?
Provide Devfile v2 for che-idea editor configuration.

**_Prerequisites for merging this PR:_**

- [x] **_Repository with sample [gradle-demo-project](https://github.com/che-samples/gradle-demo-project/tree/devfilev2) should be forked to [crw-samples](https://github.com/crw-samples/)_**

Aligned editor id in both plugin-registries. So there is no need to update forked repository to use different editor from CRW plugin-registry.

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-2686




### How to test

- Request OCP cluster on RHPDS and install CRW operator from tech-preview channel. Also create the Che cluster.

> As far as it will be CRW 2.14.0 and it's container registry does not contain:
> ```
> registry.redhat.io/codeready-workspaces/idea-rhel8:2.16
> ```
> we will use `idea-rhel8:2.14` as tested version. There isn't any changes in those editor container. 
> So our tested image will use the following digest:
> ```
> registry.redhat.io/codeready-workspaces/idea-rhel8@sha256:073d1a803faac261bae15642e66213a54cf2cb96429254c1138b0eda3f182105
> ```

- Create the workspace from the following repository: https://github.com/vzhukovs/demo-repository/tree/crw-2686
  - It contains [the same editor configuration](https://github.com/vzhukovs/demo-repository/blob/crw-2686/.che/che-editor.yaml) as in the following PR besides the editor container version.

- Make sure that workspace started.

#### Release Notes
Provide Devfile v2 for che-idea editor configuration.


#### Docs PR (if applicable)
